msticpy/context/tiproviders/mblookup.py

```diff
@@ -0,0 +1,317 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# Author: Thomas Roccia - @fr0gger_
+# --------------------------------------------------------------------------
+"""
+MalwareBazaar Provider.
+
+Input can be a hash, tag, signature, filetype, clamav signature, imphash, dhash,
+tlsh, telfhash, certificate issuer info, certificate subject info, or certificate.
+
+The class can be used to:
+    - Get data about a sample from MalwareBazaar
+    - Download malware sample from MalwareBazaar
+    - Query Code Signing Certificate Blocklist (CSCB)
+    - Get the recent addition from MalwareBazaar
+
+"""
+from enum import Enum
+from typing import Set
+
+import json
+import requests
+import pandas as pd
+
+from pandas import json_normalize
+from ..._version import VERSION
+from ...common.provider_settings import get_provider_settings
+
+__version__ = VERSION
+__author__ = "Thomas Roccia | @fr0gger_"
+
+_BASE_URL = "https://mb-api.abuse.ch/api/v1/"
+
+class MBEntityType(Enum):
+    """MBEntityType: Enum class for MalwareBazaar entity types."""
+
+    HASH = "hash"
+    TAG = "tag"
+    SIGNATURE = "signature"
+    FILETYPE = "filetype"
+    CLAMAV = "clamav"
+    IMPHASH = "imphash"
+    DHASH = "dhash"
+    YARA = "yara"
+    TLSH = "tlsh"
+    TELFHASH = "telfhash"
+    CODESIGNISSUER = "issuerinfo"
+    CODESIGNSUBJECT = "subjectinfo"
+    CODESIGNSN = "certificate"
+    GIMPHASH = "gimphash"
+
+class MBlookup:
+    """ MBlookup Python Class wrapper for MalwareBazaar API.
+
+        Classes:
+            MBLookup: Class for MalwareBazaar API.
+
+        Functions:
+            lookup_ioc(self, ioc_type: MBEntityType, ioc_value: str) -> pandas.DataFrame
+            download_sample(self, sha2: str) -> str
+            get_recent(self, selector: str) -> pandas.DataFrame
+            get_cscb(self) -> pandas.DataFrame
+    """
+
+    _SUPPORTED_MB_TYPES: Set[MBEntityType] = {
+        MBEntityType.HASH,
+        MBEntityType.TAG,
+        MBEntityType.SIGNATURE,
+        MBEntityType.FILETYPE,
+        MBEntityType.CLAMAV,
+        MBEntityType.IMPHASH,
+        MBEntityType.DHASH,
+        MBEntityType.YARA,
+        MBEntityType.TLSH,
+        MBEntityType.TELFHASH,
+        MBEntityType.CODESIGNISSUER,
+        MBEntityType.CODESIGNSUBJECT,
+        MBEntityType.CODESIGNSN,
+        MBEntityType.GIMPHASH,
+    }
+
+    def __init__(self, mb_key=None):
+        """Init function to get the API key if necessary"""
+        self.mb_key = mb_key # or_get_mb_api_key()

```

Comment:
is the or statement meant to be commented out?

Reply:
Actually, for the current version there is no need to have the API I just let this piece of code for further development